### PR TITLE
Bash script to automate development environment setup

### PR DESCRIPTION
**Pull request by gurashish1singh**
This is setting up a template repo so that I do not have to copy over config files through new repos.

The script will:
- Install poetry 
- Setup the poetry env without user intervention.
- It will add the linter packages as dev dependencies.
- Install the pre-commit hook (most of the hooks will only run during commit and some will only run during push)

I'm also shipping a base logging.yaml file and the following third party packages to be installed by default:
- python-json-logger
- pyyaml
- flake8
- black
- isort
- pre-commit

**NOTE**: For now this works on Gitbash. I tried testing in WSL and it worked in that environment as well but Poetry was installed using 3.8 (default in WSL Ubuntu). So I will create issues to try and install poetry with 3.9 and will also create one for Powershell.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        # curl -sSL https://install.python-poetry.org | python -
+    fi
+
+    msg "Initializing a new poetry project"
+    poetry init
+    echo
+}
+
+install_hooks()
+{
```
Comment (seanshir) [inline setup.sh:38]: this is not being closed anywhere yet?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[flake8]
+exclude =
+    # No need to traverse our git directory
+    .git,
+    # There's no value in checking cache directories
+    __pycache__
+max_line_length = 100
```
Comment (seanshir) [inline setup.cfg:9]: 120? 😄

**Comment by jim-sheldon** [comment]
HELLO

**Comment by gurashish1singh** [comment]
> HELLO

Oh hey Jim! ;)

File: setup.cfg (hunk 2 above)
Comment (gurashish1singh) [reply setup.cfg:9]: Nah 100 is better

File: setup.sh (hunk 1 above)
Comment (gurashish1singh) [reply setup.sh:38]: 🤔 It's umm closed

**Pull request merged**